data_validation/validation_builder.py

```diff
@@ -384,11 +384,10 @@ def get_source_query(self):
             "source_query": self.config_manager.source_query,
         }
 
-        if self.validation_type == consts.CUSTOM_QUERY:
-            table = self.config_manager.get_source_ibis_table_from_query()
-        else:
+        if self.config_manager.source_table:
```

Comment:
Same as above - why do we need to check if source_table exists rather than keeping `if self.validation_type == consts.CUSTOM_QUERY:`

Reply:
Yes, you are correct. It turns out this code is not called by generate-partitions at all, it is only called when doing a dry-run or executing the query, so I need not have made this change.